From 71d56dd546f8e27fa7a4d0e4f8bb30bd722a656a Mon Sep 17 00:00:00 2001
From: Mutwin Kraus <mutle@blogage.de>
Date: Wed, 27 Aug 2008 12:54:30 +0200
Subject: [PATCH] Sending would fail before, when a return-path header is set.
 This strips out additional < and > when setting return-path as the
 sender address

---
 actionmailer/lib/action_mailer/base.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionmailer/lib/action_mailer/base.rb b/actionmailer/lib/action_mailer/base.rb
index 72c9452..de79dda 100644
--- a/actionmailer/lib/action_mailer/base.rb
+++ b/actionmailer/lib/action_mailer/base.rb
@@ -627,7 +627,7 @@ module ActionMailer #:nodoc:
       def perform_delivery_smtp(mail)
         destinations = mail.destinations
         mail.ready_to_send
-        sender = mail['return-path'] || mail.from
+        sender = mail['return-path'] ? mail['return-path'].gsub(/^<(.*)>$/, '\1') : mail.from
 
         Net::SMTP.start(smtp_settings[:address], smtp_settings[:port], smtp_settings[:domain],
             smtp_settings[:user_name], smtp_settings[:password], smtp_settings[:authentication]) do |smtp|
-- 
1.5.5.3

